src/main/scala/beam/utils/data/synthpop/models/Models.scala

```diff
@@ -40,7 +40,7 @@ object Models {
   }
 
   case class BlockGroupGeoId(state: State, county: County, tract: String, blockGroup: String) extends GenericGeoId {
-    def asUniqueKey: String = s"${state.value}-${county.value}-$tract-$blockGroup"
+    def asUniqueKey: String = s"${state.value}${county.value}$tract$blockGroup"
```

Comment:
Will this create compatibility issues? Why make the change?

Reply:
So it looks like I made this change to make the unique `Long` value. At the time, I had little understanding of urbansim_v2 and I reviewed [this csv](https://github.com/LBNL-UCB-STI/beam-data-austin/blob/develop/urbansim_v2/blocks.csv.gz) to get more context. The original hyphenated string is used in `PumaLevelScenarioGenerator` which I'm not familiar with. The `ScenarioGenerator` class is our focus.

I think we can revert this and the type of a value should not be the problem, I'll try to generate a plan using the old format and see if it causes any problem or not to run simulation with. I'll get back to you on this one.

Reply:
Found the reason. The `UrbansimReaderV2` expects block id to be a long value. Following are important symbols,
`beam.utils.scenario.urbansim.censusblock.entities.InputHousehold#blockId: Long`
`beam.utils.scenario.urbansim.censusblock.entities.Block#blockId: Long`

Error logs:
```
11:46:10.746 INFO  b.u.s.u.censusblock.UrbansimReaderV2 - Households info has been read successfully.
Exception in thread "main" java.lang.NumberFormatException: For input string: "53-033-006800-1"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Long.parseLong(Long.java:589)
```

Again we can do one of the following two things:
1. add another method/value to return long value of unique key
2. change above two mentioned symbols to be a string which will not be a breaking change (see following changes)

```diff
diff --git a/src/main/scala/beam/utils/scenario/urbansim/censusblock/entities/Block.scala b/src/main/scala/beam/utils/scenario/urbansim/censusblock/entities/Block.scala
index d53b79dcb..be735028d 100644
--- a/src/main/scala/beam/utils/scenario/urbansim/censusblock/entities/Block.scala
+++ b/src/main/scala/beam/utils/scenario/urbansim/censusblock/entities/Block.scala
@@ -5,7 +5,7 @@ import java.util
 import beam.utils.scenario.urbansim.censusblock.EntityTransformer
 
 case class Block(
-  blockId: Long,
+  blockId: String,
   x: Double,
   y: Double
 )
@@ -13,7 +13,7 @@ case class Block(
 object Block extends EntityTransformer[Block] {
 
   override def transform(rec: util.Map[String, String]): Block = {
-    val blockId = getIfNotNull(rec, "block_id").toLong
+    val blockId = getIfNotNull(rec, "block_id")
     val x = getIfNotNull(rec, "x").toDouble
     val y = getIfNotNull(rec, "y").toDouble
 
diff --git a/src/main/scala/beam/utils/scenario/urbansim/censusblock/entities/InputHousehold.scala b/src/main/scala/beam/utils/scenario/urbansim/censusblock/entities/InputHousehold.scala
index 127e9373d..2c0dc0787 100644
--- a/src/main/scala/beam/utils/scenario/urbansim/censusblock/entities/InputHousehold.scala
+++ b/src/main/scala/beam/utils/scenario/urbansim/censusblock/entities/InputHousehold.scala
@@ -8,7 +8,7 @@ case class InputHousehold(
   householdId: String,
   income: Int,
   cars: Int,
-  blockId: Long
+  blockId: String
 )
 
 object InputHousehold extends EntityTransformer[InputHousehold] {
@@ -17,7 +17,7 @@ object InputHousehold extends EntityTransformer[InputHousehold] {
     val householdId = getIfNotNull(rec, "household_id")
     val income = getIfNotNull(rec, "income").toInt
     val cars = getIfNotNull(rec, "cars").toInt
-    val blockId = getIfNotNull(rec, "block_id").toLong
+    val blockId = getIfNotNull(rec, "block_id")
 
     InputHousehold(householdId, income, cars, blockId)
   }
diff --git a/src/main/scala/beam/utils/scenario/urbansim/censusblock/merger/HouseholdMerger.scala b/src/main/scala/beam/utils/scenario/urbansim/censusblock/merger/HouseholdMerger.scala
index e0342ce2e..291ca3a8d 100644
--- a/src/main/scala/beam/utils/scenario/urbansim/censusblock/merger/HouseholdMerger.scala
+++ b/src/main/scala/beam/utils/scenario/urbansim/censusblock/merger/HouseholdMerger.scala
@@ -4,7 +4,7 @@ import beam.sim.population.PopulationAdjustment
 import beam.utils.scenario.urbansim.censusblock.entities.{Block, InputHousehold}
 import beam.utils.scenario.{HouseholdId, HouseholdInfo}
 
-class HouseholdMerger(blocks: Map[Long, Block]) extends Merger[InputHousehold, HouseholdInfo] {
+class HouseholdMerger(blocks: Map[String, Block]) extends Merger[InputHousehold, HouseholdInfo] {
   override def merge(iter: Iterator[InputHousehold]): Iterator[HouseholdInfo] = iter.map(inputToOutput)
 
   private def inputToOutput(inputHousehold: InputHousehold): HouseholdInfo = {
```

Reply:
Interesting - I have no opinion then. As long as it doesn't break compatibility...which it sounds like it only HELPS compatibility.